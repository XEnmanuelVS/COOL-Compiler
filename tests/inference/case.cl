class C {
    
};

class B inherits C {
};

class A inherits C {
};

class Main inherits IO {
    func(): AUTO_TYPE {
        case 1 + 1 of
            a: A => new A;
            b: B => new B;
        esac
    };
};
