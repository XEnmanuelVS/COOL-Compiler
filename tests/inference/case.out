\__ProgramNode [<class> ... <class>]
  \__ClassDeclarationNode: class C  { <feature> ... <feature> }

  \__ClassDeclarationNode: class B : C { <feature> ... <feature> }

  \__ClassDeclarationNode: class A : C { <feature> ... <feature> }

  \__ClassDeclarationNode: class Main : IO { <feature> ... <feature> }
    \__FuncDeclarationNode: func() : C -> <body>
      \__CaseNode: case <expr> of [<case> ... <case>] esac
        \__case 
          \__<expr> PlusNode <expr>
            \__IntegerNode: 1
            \__IntegerNode: 1 of
      \__CaseNode: a : A =>
            \__InstantiateNode: new A()
      \__CaseNode: b : B =>
            \__InstantiateNode: new B()
